python/dicebeard/__init__.py

```diff
@@ -1,48 +1,116 @@
-import logging
-import dice
-import re
+import os
+
+from . import image_dice as dice
+from . import image_coin as coin
+
+from pathlib import Path
 
 import telepot
 import telepot.aio
-from skybeard.beards import BeardChatHandler
+from skybeard.utils import get_args
+from telepot import glance
+from telepot.namedtuple import InlineKeyboardMarkup, InlineKeyboardButton
+from skybeard.beards import BeardChatHandler, ThatsNotMineException
 
-logger = logging.getLogger(__name__)
+#logger = logging.getLogger(__name__)
 
+class DiceBeard(BeardChatHandler):
 
-def get_args(msg_text):
-    return msg_text.split(" ")[1:]
+    __commands__ = [
+        ('roll', 'roll',
+         'Rolls dice. Parses args and rolls.'),
+        ('flip', 'flip_coin',
+         'Flips a number of coins and returns the result'),
+        ('mode', 'mode',
+         'Can change the output mode of the bot between picture, icons and text'),
+    ]
 
+    __userhelp__ = """Can roll dice or flip coins. 
+    To roll dice use the /roll command followed by any number of arguments of the form 3d6+5 (can be + or -) seperated by spaces.
+    Currently, supported dice for producing images are d4, d6, d8, d10, d12 and d20.
+    To flip coins simply type /flip followed by the number of coins you would like to flip (e.g /flip 10 will filp 10 coins)"""
+          
+    images_path = Path(os.path.dirname(__file__)) / 'images'
+    my_dice = dice.Dice(images_path)
+    my_coin = coin.Coin(images_path)
+ 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.keyboard = InlineKeyboardMarkup(
+            inline_keyboard=[
+                [InlineKeyboardButton(
+                    text='Picture', callback_data=self.serialize('Picture')),
+                 InlineKeyboardButton(
+                     text='Text', callback_data=self.serialize('Text'))],
+            ])
+    
+        
+    async def roll(self, msg):
 
-dice_faces = {
-    1: "\u2680",
-    2: "\u2681",
-    3: "\u2682",
-    4: "\u2683",
-    5: "\u2684",
-    6: "\u2685",
-}
+        input_args = get_args(msg, return_string = True)        
+        out_dice = self.my_dice.roll_dice(input_args)
+        
+        #Check which mode the user is in and output the correct format
+        if self.my_dice.mode == 'pic':
+            #Check a file path has been output, output picture if so and text otherwise
+            if isinstance(out_dice, Path):
+                await self.sender.sendPhoto(open(str(out_dice),'rb'))
```

Comment:
`Path`'s have open methods which you may find useful: https://docs.python.org/3.5/library/pathlib.html#pathlib.Path.open

A more pythonic way to do this would be:
```python
try:
    await self.sender.sendPhoto(out_dice.open('rb'))
except AttributeError:
    await self.sender.sendPhoto(out_dice)
```
(which roughly translates as 'try and send it as a `Path`, but if it's not a `Path`, try sending at a string', part of Python's [EAFP principle](https://stackoverflow.com/questions/11360858/what-is-the-eafp-principle-in-python)).